frame/contracts/src/weights.rs

```diff
@@ -153,1251 +153,1321 @@ pub trait WeightInfo {
+			.saturating_add((70_558_000 as Weight).saturating_mul(n as Weight))
 			.saturating_add(T::DbWeight::get().reads(6 as Weight))
 			.saturating_add(T::DbWeight::get().writes(2 as Weight))
 	}
 	fn seal_clear_storage(r: u32, ) -> Weight {
 		(0 as Weight)
-			// Standard Error: 2_745_000
-			.saturating_add((2_316_433_000 as Weight).saturating_mul(r as Weight))
+			// Standard Error: 2_762_000
+			.saturating_add((1_268_979_000 as Weight).saturating_mul(r as Weight))
 			.saturating_add(T::DbWeight::get().reads(5 as Weight))
 			.saturating_add(T::DbWeight::get().reads((100 as Weight).saturating_mul(r as Weight)))
 			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 			.saturating_add(T::DbWeight::get().writes((100 as Weight).saturating_mul(r as Weight)))
 	}
 	fn seal_get_storage(r: u32, ) -> Weight {
-		(87_218_000 as Weight)
-			// Standard Error: 745_000
-			.saturating_add((948_121_000 as Weight).saturating_mul(r as Weight))
+		(13_033_000 as Weight)
+			// Standard Error: 1_050_000
+			.saturating_add((928_085_000 as Weight).saturating_mul(r as Weight))
 			.saturating_add(T::DbWeight::get().reads(5 as Weight))
 			.saturating_add(T::DbWeight::get().reads((100 as Weight).saturating_mul(r as Weight)))
+			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
 	fn seal_get_storage_per_kb(n: u32, ) -> Weight {
-		(719_050_000 as Weight)
-			// Standard Error: 266_000
-			.saturating_add((154_812_000 as Weight).saturating_mul(n as Weight))
+		(621_971_000 as Weight)
+			// Standard Error: 449_000
+			.saturating_add((155_007_000 as Weight).saturating_mul(n as Weight))
 			.saturating_add(T::DbWeight::get().reads(6 as Weight))
+			.saturating_add(T::DbWeight::get().writes(1 as Weight))
 	}
 	fn seal_transfer(r: u32, ) -> Weight {
-		(19_439_000 as Weight)
-			// Standard Error: 2_468_000
-			.saturating_add((5_674_822_000 as Weight).saturating_mul(r as Weight))
+		(0 as Weight)
+			// Standard Error: 1_989_000
+			.saturating_add((5_346_330_000 as Weight).saturating_mul(r as Weight))
 			.saturating_add(T::DbWeight::get().reads(5 as Weight))
 			.saturating_add(T::DbWeight::get().reads((100 as Weight).saturating_mul(r as Weight)))
-			.saturating_add(T::DbWeight::get().writes(1 as Weight))
+			.saturating_add(T::DbWeight::get().writes(2 as Weight))
 			.saturating_add(T::DbWeight::get().writes((100 as Weight).saturating_mul(r as Weight)))
 	}
 	fn seal_call(r: u32, ) -> Weight {
 		(0 as Weight)
-			// Standard Error: 7_465_000
-			.saturating_add((11_066_530_000 as Weight).saturating_mul(r as Weight))
+			// Standard Error: 11_341_000
+			.saturating_add((11_820_334_000 as Weight).saturating_mul(r as Weight))
 			.saturating_add(T::DbWeight::get().reads(6 as Weight))
 			.saturating_add(T::DbWeight::get().reads((200 as Weight).saturating_mul(r as Weight)))
+			.saturating_add(T::DbWeight::get().writes(1 as Weight))
+			.saturating_add(T::DbWeight::get().writes((100 as Weight).saturating_mul(r as Weight)))
```

Comment:
The write to the contract info data structure now billed on the `call` (100 writes per `r` means 1 write per call) rather than on contract modifying setter functions.